pkg/controller/utils/utils.go

```diff
@@ -161,3 +153,29 @@ const (
 func GetConcurrentReconciles() int {
 	return concurrentControllerReconciles
 }
+
+// MergeJsons will merge the global and local pull secret and return it
+// TODO Return better errors
+func MergeJsons(globalPullSecret string, localPullSecret string) (string, error) {
+	var mGlobal, mLocal DockerConfigJSON
+	jGlobal := []byte(globalPullSecret)
+	err := json.Unmarshal(jGlobal, &mGlobal)
+	if err != nil {
+		return "", err
+	}
+
+	jLocal := []byte(localPullSecret)
+	err = json.Unmarshal(jLocal, &mLocal)
+	if err != nil {
+		return "", err
+	}
+
+	for k, v := range mLocal.Auths {
+		mGlobal.Auths[k] = v
```

Comment:
I think it would be helpful to do an info message when the local auth overrides a global auth.

Reply:
What do you mean by "config comment"? It would be good to see something at INFO level in the logs when we do merge pull secrets.

Reply:
I am going to add an info as you both think that way.

Reply:
ok, I still do not sure about it. As in the SRE use case or in general we will not overwrite the auth. Because the regsirty names are different (auth contains registry names) 

In the current code I added a check to see if are overwriting. Let me know your opinion on this.

Reply:
The info message should indicate which specific auth was overwritten.